omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```
@@ -0,0 +1,199 @@
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	/**
+	 * @return the TagService
+	 */
+	private TagService getService() {
+		return Context.getService(TagService.class);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("tag");
+		description.addRequiredProperty("objectType");
+		description.addRequiredProperty("objectUuid");
+		
+		return description;
+	}
+	
+	/**
+	 * @see DelegatingCrudResource#doGetAll(RequestContext)
+	 */
+	@Override
+	protected NeedsPaging<Tag> doGetAll(RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	private String getUriOfObject(String objectType, String objectUuid) {
+		Class<?> className;
+		OpenmrsObject openmrsObject = null;
+		try {
+			className = OpenmrsClassLoader.getInstance().loadClass(objectType);
+			openmrsObject = (OpenmrsObject) className.newInstance();
+		}
+		catch (ClassNotFoundException e) {}
+		catch (IllegalAccessException e) {}
+		catch (InstantiationException e) {
+			e.printStackTrace();
+		}
+		openmrsObject.setUuid(objectUuid);
+		String Uri = Context.getService(RestService.class)
+		        .getResourceByName("v1/" + objectType.substring(objectType.lastIndexOf(".") + 1).toLowerCase())
+		        .getUri(openmrsObject);
+		
+		return Uri;
+	}
+	
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+		String tagName = context.getRequest().getParameter("tag");
+		String objectType = context.getRequest().getParameter("objectType");
+		String objectUuid = context.getRequest().getParameter("objectUuid");
+		if (StringUtils.isNotBlank(objectType)) {
```

You need to throw an exception if objectType is null